src/main/java/org/mule/service/http/impl/service/server/grizzly/ResponseCompletionHandler.java

```diff
@@ -155,7 +157,8 @@ public void cancelled() {
   @Override
   public void failed(Throwable throwable) {
     super.failed(throwable);
-    responseStatusCallback.onErrorSendingResponse(throwable);
+    responseStatusCallback.onErrorSendingResponse(ctx.getConnection().isOpen() ? throwable
+        : new ConnectionException(throwable));
```

Comment:
add the message to this exception in a way consistent with `ResponseStreamingCompletionHandler`, extract tha message to a constant